src/pages/studyView/chartHeader/ChartHeader.tsx

```diff
@@ -40,7 +40,7 @@ export interface IChartHeaderProps {
     chartControls?   : ChartControls;
     changeChartType  : (chartType: ChartType) => void;
     getSVG?          : ()=>Promise<SVGElement | null>;
-    getData?         : (dataType?:DataType)=>Promise<string | null>;
+    getData?         : ((dataType?:DataType)=>Promise<string | null>) | ((dataType?:DataType)=>string);
```

Comment:
i'm curious why?   seems like code will now have to check whether it's a promise or not, no?

Reply:
Not sure what's the best way here. There are some getData methods don't require a promise to produce the data. If we don't provide the second type, then these methods have to attach an async tag.